src/types/migration/Asset.ts

```diff
@@ -0,0 +1,140 @@
+import type { Asset } from "../api/asset/asset"
+import type { FilledImageFieldImage } from "../value/image"
+import type { LinkToMediaField } from "../value/linkToMedia"
+import { type RTImageNode } from "../value/richText"
+
+import type { InjectMigrationSpecificTypes } from "./Document"
+
+/**
+ * An asset to be uploaded to Prismic media library.
+ */
+export type MigrationAssetConfig = {
+	/**
+	 * ID the assets is indexed with on the migration instance.
+	 */
+	id: string | URL | File | NonNullable<ConstructorParameters<File>[0]>[0]
+
+	/**
+	 * File to be uploaded as an asset.
+	 */
+	file: string | URL | File | NonNullable<ConstructorParameters<File>[0]>[0]
```

Comment:
❓ #ask: Why do we have the same type signature for `id` and `file`? Do we need both?

Reply:
When migrating from Prismic documents, existing images and links to media are indexed on their asset ID to allow for deduplication of them if they appear multiple times which is different from their file URL.

When creating an asset regularly, we instead index on the file which is indeed the same.

I'll add a comment to clarify that.